docs/howtos/lr_schedule.rst

```diff
@@ -1,138 +1,158 @@
 Learning Rate Scheduling
 =============================
 
-Note: See "Learning Rate Schedules" in :doc:`optax_update_guide` for
-implementing learning rate schedules using ``optax``.
-
-
 The learning rate is considered one of the most important hyperparameters for
 training deep neural networks, but choosing it can be quite hard.
-To simplify this, one can use a so-called *cyclic learning rate*, which
-virtually eliminates the need for experimentally finding the best value and
-schedule for the global learning rate. Instead of monotonically decreasing the
-learning rate, this method lets the learning rate cyclically vary between
-reasonable boundary values.
-Here we will show you how to implement a triangular learning rate scheduler,
-as described in the paper  `"Cyclical Learning Rates for Training Neural Networks" <https://arxiv.org/abs/1506.01186>`_.
+Rather than simply using a fixed learning rate, it is common to use a learning rate scheduler.
+In this example, we will use the *cosine scheduler*.
+Before the cosine scheduler comes into play, we start with a so-called *warmup* period in which the
+learning rate increases linearly for ``warmup_epochs`` epochs.
+For more information about the cosine scheduler, check out the paper
+`"SGDR: Stochastic Gradient Descent with Warm Restarts" <https://arxiv.org/abs/1608.03983>`_.
 
 We will show you how to...
 
 * define a learning rate schedule
 * train a simple model using that schedule
 
-The triangular schedule makes your learning rate vary as a triangle wave during training, so over the course of a period (``steps_per_cycle``
-training steps) the value will start at ``lr_min``, increase linearly to ``lr_max``, and then decrease again to ``lr_min``.
 
 .. testsetup::
 
   import jax
+  import jax.numpy as jnp
+  from flax.training import train_state
+  import optax
+  import functools
 
 .. testcode::
   
-  def create_triangular_schedule(lr_min, lr_max, steps_per_cycle):
-    top = (steps_per_cycle + 1) // 2
-    def learning_rate_fn(step):
-      cycle_step = step % steps_per_cycle
-      if cycle_step < top:
-        lr = lr_min + cycle_step/top * (lr_max - lr_min)
-      else:
-        lr = lr_max - ((cycle_step - top)/top) * (lr_max - lr_min)
-      return lr
-    return learning_rate_fn
-
-
-To use the schedule, one must create a learning rate function by passing the hyperparameters to the
-create_triangular_schedule function and then use that function to compute the learning rate for your updates.
-For example using this schedule on MNIST would require changing the train_step function
+  def create_learning_rate_fn(config, base_learning_rate, steps_per_epoch):
+    """Create learning rate schedule."""
```

Comment:
nit: use descriptive-style in function pydoc:
https://google.github.io/styleguide/pyguide.html#383-functions-and-methods